scripts/create_changelog.py

```diff
@@ -0,0 +1,343 @@
+        ' - github = github.com account name',
+        ' - gitlab = gitlab.com account name',
+        '',
+        'Please keep the list in alphabetical order by last name!',
+        '-->',
+        '<contributors>'
+    ]
+
+    # Write contributors with precise formatting
+    for contributor in sorted_contributors:
+        contrib_line = '<contributor'
+        for attr, value in contributor.attrib.items():
+            contrib_line += f'\n {attr}="{html.escape(value,quote=False)}"'
+        contrib_line += '/>'
+        xml_lines.append(contrib_line)
+    xml_lines.append('</contributors>')
+    
+    with open(original_file, 'w',encoding='utf-8') as f:
+        f.write('\n'.join(xml_lines))
+
+
+def map_git_to_names():
+    tree = ET.parse('conf/contributors.xml')
+    root = tree.getroot()
+    for c in root.findall('contributor'):
+        name = c.get('name')
+        git = c.get('github')
+        if git and name:
+            git_to_name[git] = unidecode(name)
+
+
+def fetch_real_name(github_name):
+    print(github_name)
+    url = f"https://api.github.com/users/{github_name}"
+    try:
+        res = requests.get(url,headers=HEADERS)
+        res.raise_for_status()
+        user = res.json()
+        if not user['name']:
+            return
+        name_parts = user['name'].split()
+        if len(name_parts) < 2: # Full name is not available
+            return
+        git_to_name[github_name] = user['name']
+        new_names.append((user['name'],github_name))
+    except Exception as e:
+        print(f"Failed to fetch real name for @{github_name} : {str(e)}")
+
+
+def update_names():
+    """
+    Replace the github usernames with real names. If name is not found in contributors.xml, 
+    then github usernames are used in the form @<github-username>
+    """
+    global all_contribs
+    global first_contribs
+    for c in all_contribs:
+        if c not in git_to_name:
+            fetch_real_name(c)
+    for tag in all_info:
+        for pr in all_info[tag]:
+            pr['creator'] = git_to_name.get(pr['creator'],f"@{pr['creator']}")
+            pr['authors'] = [git_to_name.get(a,f"@{a}")  for a in pr['authors']]
+            pr['reviewers'] = [git_to_name.get(r,f"@{r}") for r in pr['reviewers']]
+    all_contribs = set([git_to_name.get(c,f"@{c}") for c in all_contribs])
+    first_contribs = set([git_to_name.get(c,f"@{c}") for c in first_contribs])
+
+
+def get_release_data(tag):
+    url = fr"{BASE_URL}/releases/tags/{tag}"
+    res = requests.get(url, headers=HEADERS)
+    if res.status_code == 404:
+        print(f"{tag} release not found")
+        return None
+    if res.status_code != 200:
+        print(f"Failed to fetch release data: {res.status_code}")
+        return None
+    return res.json()
+
+
+def get_release_date(release_data):
+    if not release_data:
+        return 'N/A'
+    date_time = release_data.get('published_at', '')
```

Comment:
If `release_data` is `None`, this function should not be called in the first place. Perhaps this `if` statement should be moved to the caller?

Reply:
It was there for just testing purpose, just so that we can test on 10.5 (Since the 10.5 release is not out yet, but there are few beta's and rc's). Will remove it